src/Hyperion/SerializerFactories/ArraySerializerFactory.cs

```diff
@@ -30,12 +32,13 @@ private static void WriteValues<T>(T[] array, Stream stream, Type elementType, V
                 stream.WriteObject(value, elementType, elementSerializer, preserveObjectReferences, session);
             }
         }
-        private static void ReadValues<T>(int length, Stream stream, DeserializerSession session, T[] array)
+        
+        private static void ReadValues(int length, Stream stream, DeserializerSession session, Array array)
```

Comment:
Need to change generic type to untyped Array array. Generic typed array does not play nice with user defined structs.